src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -590,9 +590,9 @@ static void initOutput(PrintStream ps) {
         ostream = ps;
     }
 
-    static String getMainClassFromJar(String jarname) {
+    private static String getMainClassFromJar(String jarname, JarFile jarFile) {
         String mainValue;
-        try (JarFile jarFile = new JarFile(jarname)) {
+        try {
             Manifest manifest = jarFile.getManifest();
```

Comment:
I think the try-catch around the block can be dropped and you can put a more targeted try-catch around getManifest, at least I think that is the only case remaining in getMainClassFromJar that needs error handling now.

Reply:
Dropping that outer try-catch will lead to many whitespace-only changes due to un-indenting lines of the former block. Proceed or keep the indentation stable by making that internal method throw IOE and insert a comment-only block like:

```java
/* keep indentation stable */ {
    Manifest manifest = jarFile.getManifest();
// ...
}    
```

Reply:
> Dropping that outer try-catch will lead to many whitespace-only changes due to un-indenting lines of the former block. Proceed or keep the indentation stable by making that internal method throw IOE and insert a comment-only block like:
> 
> ```java
> /* keep indentation stable */ {
>     Manifest manifest = jarFile.getManifest();
> // ...
> }    
> ```

I think it's okay to shift-tab to re-align it after you remove the try-catch. It's a small patch and the changes are easy to see.